hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestRMStoreCommands.java

```diff
@@ -101,6 +102,16 @@ public void testFormatConfStoreCmdForZK() throws Exception {
     }
   }
 
+  @Test
+  public void testSSLEnabledConfiguration() {
+    //Test if we can enable SSL/TLS for the ZK Curator Client in YARN.
+    Configuration conf = new Configuration();
+    conf.set(YarnConfiguration.RM_ZK_CLIENT_SSL_ENABLED, Boolean.TRUE.toString());
+
+    assertEquals("The " + YarnConfiguration.RM_ZK_CLIENT_SSL_ENABLED + " value should be true.",
+        conf.get(YarnConfiguration.RM_ZK_CLIENT_SSL_ENABLED), Boolean.TRUE.toString());
+  }
```

Comment:
Here, you are only testing the behavior of the YarnConfiguration class as you set something to it, and you assert what's coming back with `conf.get()`.
In my view, this kind of test does not belong here but to the tests of `YarnConfiguration` but I assume the `Configuration` class has a similar or same testcase for string values.

Besides, is this config gonna be a boolean-typed or string-typed config?
There's also a method called `Configuration#setBoolean`.

Can you add 2 testcases for the SSL settings? 
They should test that if the ZKCuratorManager is started with SSL disabled by default. The other test should check if the SSL setting is set to true, ZKCuratorManager is started with SSL enabled.
I'm not sure which test class those 2 testcases would belong to, but I assume there's something where the Curator classes are mocked.
If not, probably you can get away with a MockRM based tests, even if the curator classes are real instances and not mocks.
Between the two, I would prefer the former approach.

Reply:
Thank you Szilard for the review!

"The ZKCuratorManager is started with SSL disabled by default. " case is implicitly covered in the already existing TestLeaderElectorService.java, as it uses Curator.

Testing the SSL case will be more tricky due to CURATOR-658 "Add Support for TLS-enabled TestingZooKeeperMain" won't be fixed, but it seems there is a way by using ZooKeeperServerEmbeddedAdapter, which I need to explore how to implement.